projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php

```diff
@@ -351,11 +411,17 @@ public function add_gutenberg_menus( $wp_admin = false ) {
 	}
 
 	/**
-	 * Always use WP Admin for comments.
+	 * Adds Comments menu.
 	 *
 	 * @param bool $wp_admin Optional. Whether links should point to Calypso or wp-admin. Default false (Calypso).
 	 */
 	public function add_comments_menu( $wp_admin = false ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		// Force Calypso when SSO is disabled.
+		if ( $this->is_api_request && ! $this->is_sso_enabled ) {
+			parent::add_comments_menu( false );
```

Comment:
Maybe pass the condition directly to the parent::add_comments_menu instead of having a separate if?

e.g. 
`parent::add_comments_menu( ! ( $this->is_api_request && ! $this->is_sso_enabled ) )` if we would have a helper method for the condition, we could also rewrite it as
```
parent::add_comments_menu( ! $this->is_api_request_with_sso_disabled() )
```

Reply:
Yup, done in 3f35b06.